app/models/tag.rb

```diff
@@ -925,15 +947,20 @@ def remove_filter_taggings(old_filter_id=nil)
     end
   end
 
-  # Add filter taggings to this tag's works for one of its meta tags
+  # Add filter taggings to this tag's items for one of its meta tags
   def inherit_meta_filters(meta_tag_id)
     meta_tag = Tag.find_by(id: meta_tag_id)
     return unless meta_tag.present?
-    self.filtered_works.each do |work|
-      unless work.filters.include?(meta_tag)
-        work.filter_taggings.create!(inherited: true, filter_id: meta_tag.id)
-        RedisSearchIndexQueue.reindex(work, priority: :low)
-        IndexQueue.enqueue_ids(Series, work.series.pluck(:id), :background) if $rollout.active?(:start_new_indexing)
+    filtered_items = self.filtered_works + self.filtered_external_works
+    filtered_items.each do |item|
+      unless item.filters.include?(meta_tag)
+        item.filter_taggings.create!(inherited: true, filter_id: meta_tag.id)
```

Comment:
When initially creating an ExternalWork (or when editing an ExternalWork to add a new tag) no FilterTaggings are added for any meta-tags. So adding the meta-tags here means that ExternalWorks are going to have inconsistent filters: the ExternalWorks added before the meta-tag was added will have the meta-tag as a filter, but the ExternalWorks added after the meta-tag was added won't have the meta-tag as a filter.

Reply:
Sarken addressed both of your comments on a new pull request, #3326.